drivers/misc/nxp_lp_flexcomm/nxp_lp_flexcomm.c

```diff
@@ -0,0 +1,169 @@
+/*
+ * Copyright 2024 NXP
+ *
+ * SPDX-License-Identifier: Apache-2.0
+ */
+
+#define DT_DRV_COMPAT nxp_lp_flexcomm
+
+#include <errno.h>
+#include <zephyr/device.h>
+#include <zephyr/irq.h>
+#include <zephyr/kernel.h>
+#include <zephyr/logging/log.h>
+#include "nxp_lp_flexcomm.h"
+
+LOG_MODULE_REGISTER(nxp_lp_flexcomm, LOG_LEVEL_ERR);
+
+struct nxp_lp_flexcomm_child {
+	const struct device *dev;
+	uint8_t periph;
```

Comment:
do  the dev and periph fields really need to go in the RAM? device_dt_get can get the child devices at compile time and periph seems it can be determined also at compile time unless I'm mistaken.

Reply:
Looking at the driver design thats how this is needed to be.